Rubberduck.Inspections/Concrete/UnreachableCaseInspection/UnreachableCaseInspectionSelectStmt.cs

```diff
@@ -0,0 +1,202 @@
+        {
+            var caseClauseTypeNames = new List<string>();
+            foreach (var caseClause in selectStmt.caseClause())
+            {
+                foreach (var range in caseClause.rangeClause())
+                {
+                    if (TryDetectTypeHint(range.GetText(), out string hintTypeName))
+                    {
+                        caseClauseTypeNames.Add(hintTypeName);
+                    }
+                    else
+                    {
+                        var inspRange = _inspectionRangeFactory.Create(range, inspValues);
+                        var types = inspRange.ResultContexts.Select(rc => inspValues.GetTypeName(rc));
+                        caseClauseTypeNames.AddRange(types);
+                    }
+                }
+            }
+
+            if (TryDetermineEvaluationTypeFromTypes(caseClauseTypeNames, out _evalTypeName))
+            {
+                return _evalTypeName;
+            }
+            return string.Empty;
+        }
+
+        private static bool TryDetermineEvaluationTypeFromTypes(IEnumerable<string> typeNames, out string typeName)
+        {
+            typeName = string.Empty;
+            var typeList = typeNames.ToList();
+
+            //If everything is declared as a Variant, we do not attempt to inspect the selectStatement
+            if (typeList.All(tn => new string[] { Tokens.Variant }.Contains(tn)))
+            {
+                return false;
+            }
+            typeList.All(tn => new string[] { typeList.First() }.Contains(tn));
+            //If all match, the typeName is easy...This is the only way to return "String" or "Currency".
+            if (typeList.All(tn => new string[] { typeList.First() }.Contains(tn)))
+            {
+                typeName = typeList.First();
+                return true;
+            }
+            //Integer numbers will be evaluated using Long
+            if (typeList.All(tn => new string[] { Tokens.Long, Tokens.Integer, Tokens.Byte }.Contains(tn)))
+            {
+                typeName = Tokens.Long;
+                return true;
+            }
+
+            //Mix of Integertypes and rational number types will be evaluated using Double
+            if (typeList.All(tn => new string[] { Tokens.Long, Tokens.Integer, Tokens.Byte, Tokens.Single, Tokens.Double }.Contains(tn)))
+            {
+                typeName = Tokens.Double;
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool TryDetectTypeHint(string content, out string typeName)
+        {
+            typeName = string.Empty;
+            if (SymbolList.TypeHintToTypeName.Keys.Any(th => content.EndsWith(th)))
+            {
+                var lastChar = content.Substring(content.Length - 1);
+                typeName = SymbolList.TypeHintToTypeName[lastChar];
+                return true;
+            }
+            return false;
+        }
+    }
+
+    internal static class MathTokens
```

Comment:
these tokens should be made available by the Lexer already. Check if you can use those instead

Reply:
Another signpost to remind of missing operators `Imp`, `Eqv`.

Reply:
Modified MathTokens to use Tokens or Lexer values (assuming Lexer == from `VBAParser.cs`).  Same with CompareTokens where possible (Note: `<=`, `>=`, and `<>` Lexer values do not have display literals).  Changed names to MathSymbols and CompareSymbols since I'm starting to suspect 'Tokens' may not convey the right meaning.

Reply:
Operators `Imp` and `Eqv` have been added.